### PR TITLE
Brings C SDK up to date with the Rust SDK and introduces new allocation/ deallocation strategy

### PR DESCRIPTION
**Pull request by QuestofIranon**
Still needs the functions that require writing by hand

### CODE REVIEW

**Comment by QuestofIranon** [comment]
also updates to match the current rust sdk where it didn't before.

File: include/hedera-crypto.h [hunk 1]
```diff
@@ -28,7 +28,14 @@ extern HederaError hedera_signature_from_str(const char* s, HederaSignature* out
 extern char* hedera_signature_to_str(HederaSignature*);
 
 /// Generate a new [HederaSecretKey] from a cryptographically secure pseudo-random number generator (CSPRNG).
-extern HederaSecretKey hedera_secret_key_generate();
+///
```
Comment (mehcode) [inline include/hedera-crypto.h]: Random `///`

File: include/hedera-crypto.h [hunk 2]
```diff
@@ -28,7 +28,14 @@ extern HederaError hedera_signature_from_str(const char* s, HederaSignature* out
...
+///
+
+/// Generate a new [HederaSecretKey] with a BIP-39 mnemonic using a cryptographically
+/// secure random number generator.
+///
+/// The [password] is required with the mnemonic to reproduce the secret key.
+extern HederaSecretKey hedera_secret_key_generate(const char* s, const char* out);
```
Comment (mehcode) [inline include/hedera-crypto.h]: `out` -> `password`

**Comment by mehcode** [comment]
Why are claims constructible? Is that in Rust? We should remove that.

**Comment by QuestofIranon** [comment]
https://github.com/hashgraph/hedera-sdk-rust/blob/master/src/claim.rs

**Comment by mehcode** [comment]
We should definitely remove both those functions. The claims are only for returning as part of data structures.

File: src/claim.rs [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
-    account: AccountId,
-    hash: Vec<u8>
-) -> Claim {
-    Claim::new(account, hash)
-}
+use std::convert::TryInto;
+
+#[repr(C)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct CClaim(pub(crate) AccountId, pub(crate) Vec<u8>, pub(crate) Vec<PublicKey>);
```
Comment (mehcode) [inline src/claim.rs]: You can't actually return a `Vec` to C.
It has to be 1 pointer and 2 `usize`s. `capacity` and `len`.

File: src/claim.rs [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
-#[no_mangle]
-pub unsafe extern "C" fn hedera_claim_add_key(
-    claim: *mut Claim,
-    key: PublicKey
-) {
-    (&mut *claim).key(key);
+impl From<CClaim> for Claim {
+    fn from(cclaim: CClaim) -> Self {
+        Claim{
+            account: cclaim.0.try_into().unwrap(),
```
Comment (mehcode) [inline src/claim.rs]: Why `.unwrap()` Just make it a TryFrom instead of a From.

**Comment by QuestofIranon** [comment]
Don't think this is going to be ready for another day or so.

**Pull request merged**